### PR TITLE
fcitx5-sayura: init at 5.0.8

### PR DESCRIPTION
**Pull request by pluiedev**
## Description of changes
Adds the Sayura input scheme for Sinhala for Fcitx5.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Vonfry — CHANGES_REQUESTED** [review]
By the way, `nix build '.#nixosTests.fcitx5'` builds failed on my machine which is related to qt on the master. And you can add sayura into update.py to make the script work with it.

**Comment by pluiedev** [comment]
> By the way, `nix build '.#nixosTests.fcitx5'` builds failed on my machine which is related to qt on the master. And you can add sayura into update.py to make the script work with it.

So I just have to change `update.py` to make the test pass... right? Sorry I'm not particularly familiar with this workflow and I might be mistaken lol

**Review by jian-lin — CHANGES_REQUESTED** [review]
Please re-order your commits to make sure maintainer `pluiedev` is added before used.

**Review by Vonfry — APPROVED** [review]
Tested with following commands:
```
nix build --impure --expr 'with (builtins.getFlake "/home/vonfry/repos/nixpkgs").legacyPackages.x86_64-linux; fcitx5-with-addons.override { addons = [ fcitx5-sayura ]; }'
fcitx5 -r
fcitx5-configtool # add sayura

# now it worked
```

~~`nix build '.#nixosTests.fcitx5'` also successes on my machine now.~~

File: pkgs/tools/inputmethods/fcitx5/fcitx5-sayura.nix
Comment (jian-lin) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-sayura.nix:10]: This format is not align to the convention of Nixpkgs.  However, I will not request you to change this since we do not state that in the manual and do not have a formatter yet (see RFC 101).

**Review by jian-lin — APPROVED** [review]
LGTM

**Comment by NickCao** [comment]
diff LGTM, build (or test?) is failing on ofborg.

File: nixos/tests/fcitx5/default.nix
Comment (Vonfry) [inline nixos/tests/fcitx5/default.nix:144]: `machine.send_chars("g\n")` doesn't output `ග`, but `ග\n` instead. Therefore, assertion on `file_content` failed.

**Review by Vonfry — CHANGES_REQUESTED** [review]

File: nixos/tests/fcitx5/default.nix
Comment (pluiedev) [reply nixos/tests/fcitx5/default.nix:144]: Ahh... Lemme rectify thatt

**Comment by Vonfry** [comment]
You will have to rebase to master and migrate to the new fcitx5 test after #249226 is merged into master.

**Comment by pluiedev** [comment]
Closing because I don't really make use of this anymore.

**Pull request closed without merging**
